.github/workflows/test.yml

```diff
@@ -367,6 +367,8 @@ jobs:
       - test-postgres-elasticsearch7
       - test-sqlite-opensearch2
     runs-on: ubuntu-latest
+    env:
+      codecov_token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
Secrets do not get picked up on PR builds:

> With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository.

https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow

[v4.0.0](https://github.com/codecov/codecov-action/releases/tag/v4.0.0) of the codecov action had this breaking change:

> Tokenless uploading is unsupported. However, PRs made from forks to the upstream public repos will support tokenless (e.g. contributors to OS projects do not need the upstream repo's Codecov token). This [doc](https://docs.codecov.com/docs/adding-the-codecov-token#github-actions) shows instructions on how to add the Codecov token.

I believe I've set up the secrets correctly following that doc... Maybe we need to unset `fail_ci_if_error` instead?

Reply:
@laymonage Aha, that makes sense! Have just tried pushing this to a new branch of the origin repo, and it is indeed running the upload step there ([wagtail repo](https://github.com/wagtail/wagtail/actions/runs/10827138790/job/30040638418) vs [my repo](https://github.com/gasman/wagtail/actions/runs/10819530198/job/30018393366) and [PR build](https://github.com/wagtail/wagtail/actions/runs/10819530674/job/30018392977)). So maybe this fix is actually working as desired, then...

In that case, the only thing I'm not clear on is why we didn't see CI failures on PR builds before making this change - see [my previous commit running on my repo](https://github.com/gasman/wagtail/actions/runs/10818106468/job/30013832523) vs [as a PR](https://github.com/wagtail/wagtail/actions/runs/10818135211/job/30013926975), when both of them should have been missing the secret...?

Reply:
The new change doesn't seem to work, but there's a possibility that it's an issue on the action itself: https://github.com/codecov/codecov-action/issues/1548

Maybe we should just roll back to v3 for now?...

Reply:
There was a random failure on one of the postgres tests on the non-PR push that stopped it from reaching the codecov task too. Will give it one more try before admitting defeat and rolling back to v3...